ui-v2/app/services/client/http.js

```diff
@@ -165,15 +175,19 @@ export default Service.extend({
           // for write-like actions
           // potentially we should change things so you _have_ to do that
           // as doing it this way is a little magical
-          if (method !== 'GET') {
-            if (headers['Content-Type'].indexOf('json') !== -1) {
-              options.data = JSON.stringify(body);
-            } else {
-              // TODO: Does this need urlencoding? Assuming jQuery does this
-              options.data = body;
-            }
+          if (method !== 'GET' && headers['Content-Type'].indexOf('json') !== -1) {
+            options.data = JSON.stringify(body);
+          } else {
+            // TODO: Does this need urlencoding? Assuming jQuery does this
```

Comment:
Maybe this is a question to resolve before merging this?

Reply:
Oh yeah, I dropped this in just to triple check this before the entire refactor goes onto `master`. A little bit of background info here (it's a little hard to remember as its from things I did back in April):

The `client` here tries to replicate exactly what the ember-data RESTAdapter does. I'm pretty sure I copied its behaviour pretty much exactly, with a little bit of refactoring to simplify the code somewhat. As I know I did it like this I'm pretty sure I don't need to do any extra url encoding here, but you know what happens whenever you are 'pretty sure you don't need to do something' - you probably need to 😂, plus I did this back in April so I suppose I was thinking I want to double check now so much time has passed (I might be completely misremembering here) .

Anyway, this is just a little reminder for me to check - not part of this PR (I think its actually part of the previous one) but this will be checked before this entire refactor goes down onto `master`.

Actually probably worth stating here, there'll probably be a final 'cleanup' PR of this entire refactor that you'll see soonish before it goes into `master`.. Sorry.. I really need to work on my 'PR Series' fu, michael's really good at communicating those, I'm gonna try harder with that next time I have a big thing like this!